### PR TITLE
Misc rdkafka orderer updates

### PR DESCRIPTION
**Pull request by GaryWilber**
- Remove batching logic from the producer - we're all always setting `MaxBatchSize` to 1 to avoid kafka message size issues. The producer can be simplified by removing that logic
- Add a `reconnectOnNonFatalErrors` option to the producer - the librdkafka producer is smart and automatically retries message producing most of the time. It's not necessary to close and re-open it on every error it gets
- Tweak how the producer tracks the connecting / connected state
- Fix consumer errors during close / server shutdown
- Add support for disabling automatic topic creation - this logic must be disabled when targeting Azure Event Hubs
- Add support for rdkafka log events

### CODE REVIEW

File: server/routerlicious/packages/services-ordering-rdkafka/src/rdkafkaConsumer.ts [hunk 1]
```diff
@@ -263,6 +267,10 @@ export class RdkafkaConsumer extends RdkafkaBase implements IConsumer {
 			this.emit("throttled", event);
 		});
 
+		consumer.on("event.log", (event) => {
+			this.emit("log", event);
```
Comment (hedasilv) [inline server/routerlicious/packages/services-ordering-rdkafka/src/rdkafkaConsumer.ts:271]: How did you consume these `log` events? Did you add custom listeners on your RdKafkaConsumer and RdKafkaProducer objects?

File: server/routerlicious/packages/services-ordering-rdkafka/src/rdkafkaProducer.ts [hunk 2]
```diff
@@ -10,39 +10,62 @@ import {
...
 import { Deferred } from "@fluidframework/common-utils";
 
 import { IKafkaBaseOptions, IKafkaEndpoints, RdkafkaBase } from "./rdkafkaBase";
 
+/**
+ * Rdkafka producer options
+ */
 export interface IKafkaProducerOptions extends Partial<IKafkaBaseOptions> {
+	/**
+	 * Determines if the producer should be closed and reopened when a fataol error occurs.
```
Comment (hedasilv) [inline server/routerlicious/packages/services-ordering-rdkafka/src/rdkafkaProducer.ts:25]: minor neat: fatal

**Review by hedasilv — APPROVED** [review]
:shipit:

File: server/routerlicious/packages/services-ordering-rdkafka/src/rdkafkaConsumer.ts (hunk 1 above)
Comment (GaryWilber) [reply server/routerlicious/packages/services-ordering-rdkafka/src/rdkafkaConsumer.ts:271]: Yea. These log events are emitted when rdkafka debug flags are set

**Pull request merged**